src/services/chain/messages/evm/transfer.ts

```diff
@@ -58,7 +58,61 @@ export class EvmTransferMessage implements ChainMessage<SendToolParams> {
     return InProgressTxDisplay;
   }
 
-  validate(params: SendToolParams, walletStore: WalletStore): boolean {
+  private async validateBalance(
+    params: {
+      denom: string;
+      amount: string;
+      chainName: string;
+    },
+    walletStore: WalletStore,
+  ): Promise<boolean> {
+    try {
+      const { denom, amount, chainName } = params;
+      const { erc20Contracts, nativeToken, rpcUrls, nativeTokenDecimals } =
+        chainRegistry[this.chainType][chainName];
+
+      const rpcProvider = new ethers.JsonRpcProvider(rpcUrls[0]);
+      const address = walletStore.getSnapshot().walletAddress;
+
+      if (denom.toUpperCase() === nativeToken) {
+        const rawBalance = await rpcProvider.getBalance(address);
+        const formattedBalance = ethers.formatUnits(
+          rawBalance,
+          nativeTokenDecimals,
+        );
+        return Number(formattedBalance) >= Number(amount);
+      }
+
+      const erc20Record = getERC20Record(denom, erc20Contracts);
+      if (!erc20Record) {
+        return false;
+      }
+
+      const contract = new ethers.Contract(
+        erc20Record.contractAddress,
+        erc20ABI,
+        rpcProvider,
+      );
+
+      const decimals = await contract.decimals();
+      const rawBalance = await contract.balanceOf(address);
+      const formattedBalance = ethers.formatUnits(rawBalance, decimals);
+
+      console.log(
+        formattedBalance,
+        amount,
+        Number(formattedBalance) >= Number(amount),
+      );
+      return Number(formattedBalance) >= Number(amount);
+    } catch (e) {
```

Comment:
i would either remove the try/catch or rethrow the error in the catch block if we wanna create a custom error instead of the ethers library ones.